packages/expo-gl-cpp/cpp/EXGLContext.h

```diff
@@ -250,35 +256,42 @@ class EXGLContext {
 
 
 private:
-  void installMethods(JSContextRef jsCtx);
-  void installConstants(JSContextRef jsCtx);
+  void installMethods(jsi::Runtime& runtime, jsi::Object& jsGl);
+  void installConstants(jsi::Runtime& runtime, jsi::Object& jsGl);
 
-  // Utilities
-  static inline void jsThrow(JSContextRef jsCtx, const char *msg, JSValueRef *jsException) {
-    *jsException = JSValueToObject(jsCtx, EXJSValueMakeStringFromUTF8CString(jsCtx, msg), nullptr);
+  template<typename F>
+  inline jsi::Value getActiveInfo(
+          jsi::Runtime& runtime,
+          const jsi::Value* jsArgv,
+          GLenum lengthParam,
+          F &&glFunc);
+
+  template<typename T>
+  std::vector<T> jsArrayToVector(jsi::Runtime& runtime, const jsi::Array& jsArray) {
+    int length = jsArray.length(runtime);
+    std::vector<T> values(length);
+
+    for (int i = 0; i < length; i++) {
+      values[i] = static_cast<T>(jsArray.getValueAtIndex(runtime, i).asNumber());
+    }
+    return values;
   }
 
-  static inline std::shared_ptr<char> jsValueToSharedStr(JSContextRef jsCtx, JSValueRef jsVal) noexcept {
-    return std::shared_ptr<char>(EXJSValueToUTF8CStringMalloc(jsCtx, jsVal, nullptr), free);
-  }
+  template<>
```

Comment:
Is this necessary?

Reply:
it could be a separate method, but this way we can unify naming for convert functions